### ISSUE TITLE
Cannot conditionally render nodes under root template

### ISSUE THREAD

**Issue by jquesada2016** [sycamore-rs/sycamore#254]
**Describe the bug**
Not exactly sure how to word the title, some triage would be appreciated!

Currently, if performing an if/else statement in the root of a template! invocation, an unhandled exception occurs.

**To Reproduce**

use the following code:
```rust
use sycamore::{prelude::*, render};

fn main() {
    console_error_panic_hook::set_once();

    render(|| {
        let visibility = Signal::new(true);

        template! {
            button(on:click=cloned!((visibility) => move |_| visibility.set(!*visibility.get()))) {
                "Toggle Visibility"
            }
            (if *visibility.get() {
                template! {
                    h1 {
                        "I AM VISIBILE"
                    }
                }
            } else {
                template! {}
            })
        }
    });
}
```

**Expected behavior**
I expect the `<h1>` tag to be removed from the DOM without panicking nor needing a dummy wrapper node.

**Environment**

- Sycamore: v0.6.0
- Browser: Chromium Edge
- OS: Windows 10

**Additional context**
The generated error is the following:
Also note that the error can be avoided by wrapping the if/else expression in a dummy `<span>` or `<div>`

```
Uncaught RuntimeError: unreachable
    at __rust_start_panic (index-fe88caf6e9aa8c54_bg.wasm:0x7b3e4)
    at rust_panic (index-fe88caf6e9aa8c54_bg.wasm:0x78ff0)
    at std::panicking::rust_panic_with_hook::h9852a65ac22b441a (index-fe88caf6e9aa8c54_bg.wasm:0x3d655)
    at std::panicking::begin_panic_handler::{{closure}}::hb262f7ea62636fdc (index-fe88caf6e9aa8c54_bg.wasm:0x54ee9)
    at std::sys_common::backtrace::__rust_end_short_backtrace::hbeb6eae20506157b (index-fe88caf6e9aa8c54_bg.wasm:0x7b0aa)
    at rust_begin_unwind (index-fe88caf6e9aa8c54_bg.wasm:0x74dc5)
    at std::panicking::begin_panic_fmt::h8370a52944002308 (index-fe88caf6e9aa8c54_bg.wasm:0x77cb9)
    at sycamore::utils::render::reconcile_fragments::h173454c6ccadfabc (index-fe88caf6e9aa8c54_bg.wasm:0x1a86)
    at sycamore::utils::render::insert_expression::h78c7671e519664a0 (index-fe88caf6e9aa8c54_bg.wasm:0x5025)
    at sycamore::utils::render::insert_expression::{{closure}}::h9cd3308a5537a912 (index-fe88caf6e9aa8c54_bg.wasm:0x20142)
```

**Comment by arctic-hen7** [comment]
Could you try adding `console_error_panic_hook = "0.1.6"` to your `Cargo.toml` and then putting `std::panic::set_hook(Box::new(console_error_panic_hook::hook));` at the top of your your file? That will make panic messages pass through to the browser console, and you should be able to see more details.

**Comment by arctic-hen7** [comment]
I can replicate this in the Gitpod starter repo, and I'm getting this error above `unreachable executed`:

```
panicked at 'node 1 in existing nodes Vec is not a child of parent. node = DomNode(
    "No longer visible.",
)', /workspace/.cargo/registry/src/github.com-1ecc6299db9ec823/sycamore-0.6.0/src/utils/render.rs:240:17
```

**Comment by arctic-hen7** [comment]
[This](https://github.com/sycamore-rs/sycamore/blob/master/packages/sycamore/src/utils/render.rs#L243-L254) seems to be the problematic code.

```rust
    // Sanity check: make sure all nodes in a are children of parent.
    #[cfg(debug_assertions)]
    {
        for (i, node) in a.iter().enumerate() {
            if node.parent_node().as_ref() != Some(parent) {
                panic!(
                    "node {} in existing nodes Vec is not a child of parent. node = {:#?}",
                    i, node
                );
            }
        }
    }
```

**Comment by arctic-hen7** [comment]
I'm not familiar with this part of the code, so I'll defer to @lukechu10.

**Comment by lukechu10** [comment]
Should be fixed by #251 
Will create a new release soon.

**Comment by lukechu10** [comment]
Fixed in v0.6.1
